File: services/api/fortune_products.py
from dataclasses import asdict, dataclass

@dataclass(frozen=True)
class FortuneProduct:
    id: str
    name: str
    price: int
    kind: str
    summary: str
    requires_partner: bool = False

PRODUCTS = (
    FortuneProduct('daeun_deep','대운 심층 분석',9900,'time','지금 대운의 핵심 사건과 다음 전환점'),
    FortuneProduct('annual_flow','세운·연도별 분석',7900,'time','올해부터 5년의 변화와 기회'),
    FortuneProduct('monthly_calendar','월운 캘린더',5900,'time','이번 해 월별로 움직일 때와 멈출 때'),
    FortuneProduct('sinsal_detail','신살 전체 해석',5900,'detail','명식에 실제로 성립한 신살만 해석'),
    FortuneProduct('compatibility','궁합 분석 1쌍',9900,'relationship','두 명식의 끌림·충돌·지속 조건',True),
    FortuneProduct('couple_compatibility','연인·부부 심층 궁합',14900,'relationship','갈등 장면과 관계를 지키는 합의',True),
    FortuneProduct('taekil','택일 분석',7900,'detail','목적에 맞는 날짜 후보 비교'),
)
BUNDLES = (
    {'id':'time_bundle','name':'시간 흐름 패키지','price':19900,'includes':['daeun_deep','annual_flow','monthly_calendar']},
    {'id':'relationship_bundle','name':'관계 패키지','price':19900,'includes':['compatibility','couple_compatibility']},
    {'id':'all_bundle','name':'전체 분석 패키지','price':29900,'includes':[p.id for p in PRODUCTS]},
)

def public_products():
    return {'products':[asdict(p) for p in PRODUCTS], 'bundles':list(BUNDLES), 'currency':'KRW'}

def get_product(product_id):
    return next((p for p in PRODUCTS if p.id == product_id), None)

def get_offer(offer_id):
    product = get_product(offer_id)
    if product is not None: return product
    return next((b for b in BUNDLES if b['id'] == offer_id), None)

File: apps/web/lib/fortune-products.ts
export type FortuneProductId = "daeun_deep" | "annual_flow" | "monthly_calendar" | "sinsal_detail" | "compatibility" | "couple_compatibility" | "taekil";
export type FortuneProduct = { id: FortuneProductId; name: string; price: number; kind: "time" | "relationship" | "detail"; summary: string; requiresPartner?: boolean };
export const FORTUNE_PRODUCTS: FortuneProduct[] = [
 {id:"daeun_deep",name:"대운 심층 분석",price:9900,kind:"time",summary:"지금 대운의 핵심 사건과 다음 전환점"},
 {id:"annual_flow",name:"세운·연도별 분석",price:7900,kind:"time",summary:"올해부터 5년의 변화와 기회"},
 {id:"monthly_calendar",name:"월운 캘린더",price:5900,kind:"time",summary:"이번 해 월별로 움직일 때와 멈출 때"},
 {id:"sinsal_detail",name:"신살 전체 해석",price:5900,kind:"detail",summary:"명식에 실제로 성립한 신살만 해석"},
 {id:"compatibility",name:"궁합 분석 1쌍",price:9900,kind:"relationship",summary:"두 명식의 끌림·충돌·지속 조건",requiresPartner:true},
 {id:"couple_compatibility",name:"연인·부부 심층 궁합",price:14900,kind:"relationship",summary:"갈등 장면과 관계를 지키는 합의",requiresPartner:true},
 {id:"taekil",name:"택일 분석",price:7900,kind:"detail",summary:"목적에 맞는 날짜 후보 비교"}
];
export const FORTUNE_BUNDLES = [
 {id:"time_bundle",name:"시간 흐름 패키지",price:19900,includes:["daeun_deep","annual_flow","monthly_calendar"] as FortuneProductId[]},
 {id:"relationship_bundle",name:"관계 패키지",price:19900,includes:["compatibility","couple_compatibility"] as FortuneProductId[]},
 {id:"all_bundle",name:"전체 분석 패키지",price:29900,includes:FORTUNE_PRODUCTS.map(p=>p.id)}
] as const;
export function fortuneProduct(id:string){return FORTUNE_PRODUCTS.find(p=>p.id===id)??null;}
